source/non_matching/quadrature_generator.cc

```diff
@@ -1251,6 +1265,325 @@ namespace NonMatching
         AssertIndexRange(q_index, this->q_collection1D->size());
         this->q_index = q_index;
       }
+
+      DeclExceptionMsg(
+        ExcCellNotSet,
+        "The set_active_cell function has to be called before calling this function.");
+
+
+      /**
+       * This class evaluates a function defined by a solution vector and a
+       * DoFHandler transformed to reference space. To be precise, if we let
+       * $\hat{x}$ be a point on the reference cell, this class implements the
+       * function
+       *
+       * $\hat{f}(\hat{x}) = \sum_{j=0}^{n-1} f_j \hat{\phi}_j(\hat{x})$,
+       *
+       * where $f_j$ are the local solution values and $\hat{\phi}_j(\hat(x))$
+       * are the local reference space shape functions. The gradient and Hessian
+       * of this function are thus derivatives with respect to the reference
+       * space coordinates, $\hat{x}_0, \hat{x}_1, \ldots$.
+       *
+       * Note that this class is similar to FEFieldFunction, but that
+       * FEFieldFunction implements the following function on a given cell, $K$,
+       *
+       * $f(x) = \sum_{j=0}^{n-1} f_j \hat{\phi}_j(F_K^{-1}(x))$,
+       *
+       * which has the same coefficients but uses real space basis functions.
```

Comment:
Is this really a mathematically precise statement? The *polynomial basis functions* get reference space arguments in both cases, but the difference is that we plug in the reference coordinates in one case and the real coordinates in the other case. So one could see the difference as being between `\hat{phi}_j` and `\hat{\phi}_j \circ F_K^{-1}`, where the latter means the composition of different functions. I would maybe write "Note that `FEFieldFunction` gets evaluated on points in real space, i.e., the solution is expressed by the composition of the polynomial basis functions with the inverse of the mapping $F_K^{-1}$, whereas the present class directly takes reference space arguments.`